src/main/java/org/mule/extensions/jms/internal/connection/provider/activemq/ActiveMQConnectionProvider.java

```diff
@@ -35,10 +45,26 @@
  * @since 1.0
  */
 @Alias("active-mq")
-@ExternalLib(name = "ActiveMQ", description = "An ActiveMQ Broker", type = DEPENDENCY)
+@ExternalLib(name = "ActiveMQ Client", description = "The ActiveMQ JMS Client implementation.", type = DEPENDENCY,
+    requiredClassName = CONNECTION_FACTORY_CLASS, coordinates = "org.apache.activemq:activemq-client:" + ACTIVEMQ_VERSION)
+@ExternalLib(name = "ActiveMQ Broker",
+    description = "The ActiveMQ Message Broker implementation. Only required if it is required to start an on memory broker based in the VM transport.",
+    type = DEPENDENCY, requiredClassName = BROKER_CLASS, coordinates = BROKER_GA + ":" + ACTIVEMQ_VERSION,
+    optional = true)
+@ExternalLib(name = "ActiveMQ KahaDB",
+    description = "The ActiveMQ KahaDB Store Implementation. Only required if using an on memory broker and required to be persistent. E.g: 'vm://localhost?broker.persistent=true'",
```

Comment:
"Only required if using a persistent in-memory broker. For example: 'vm://localhost?broker.persistent=true'"